src/app/modules/issue-vc/new-issue-vc/new-issue-vc.component.html

```diff
@@ -6,65 +6,49 @@ <h4 mat-dialog-title class="mb-0 d-flex justify-content-between align-items-cent
 </h4>
 
 <div mat-dialog-content>
-  <div class="mb-2 mt-4 mx-4 mx-lg-4">
+  <div class="mb-2 mt-4">
     <form role="form" [formGroup]="form">
       <div class="row">
         <div class="col-lg-12">
-          <mat-label class="ml-3 pl-1">Type</mat-label>
-          <mat-form-field appearance="outline"
-                          class="mr-0 mr-lg-0 mb-2 mb-lg-0">
-            <mat-select formControlName="type" placeholder="Select Type" data-qa-id="select-type"
-                        (selectionChange)="roleTypeSelected($event)">
-              <mat-option *ngFor="let role of roles"
-                          [value]="role" [attr.data-qa-id]="role.name">{{role.name}}</mat-option>
-            </mat-select>
-          </mat-form-field>
+          <div class="card card-default d-flex shadow-none pt-3 mb-3">
+            <mat-label class="ml-3 pl-1">Subject</mat-label>
+            <mat-form-field class="mt-2" appearance="outline">
+              <input matInput formControlName="subject" placeholder="DID that is receiving the credential"
+                     maxlength="256" data-qa-id="subject-did" required autocomplete="off">
+              <button *ngIf="!isDidPredefined()" matSuffix mat-icon-button aria-label="Scan DID" appQrCodeScanner
```

Comment:
Usually is not a good approach to call method in template. I would use a getter in the .ts file and use that here:
```
get isDidPredefined() {
}

ngIf="isDidPredefined"
```

Reply:
Why it is not a good approach? I know that it is calculated on each change detection cycle, but both solutions are. There are not benefits from this. It's more like a syntactic sugar here :D

Reply:
Yeah, pipes or variables would be better. Calling methods would end up in performance issues due to change detection. So I would try to use pipes and variables where possibile. I prefer getters to calling methods mostly from a visual perspective.